src/main/java/exceptions/NoKernelException.java

```diff
@@ -0,0 +1,10 @@
+package exceptions;
```

Comment:
Check the usages on this, we might not need this now that Spring is doing all the DI for us.